src/native/corehost/host_runtime_contract.h

```diff
@@ -23,13 +23,37 @@
 #define HOST_PROPERTY_PLATFORM_RESOURCE_ROOTS "PLATFORM_RESOURCE_ROOTS"
 #define HOST_PROPERTY_TRUSTED_PLATFORM_ASSEMBLIES "TRUSTED_PLATFORM_ASSEMBLIES"
 
+struct trusted_platform_assemblies
+{
+	uint32_t assembly_count;
+	char** basenames; /* Foo.dll */
+	char** assembly_filepaths; /* /blah/blah/blah/Foo.dll */
+};
+
+struct probing_lookup_paths 
+{
+	uint32_t dir_count;
+	char** dirs;
+};
+
+struct probing_path_properties
```

Comment:
Should this rather be a callback? 

A performance flaw of the current system is that the host has to push paths to the runtime upfront. It introduces a lot of unnecessary copying and parsing. If we are redoing how this works, we should address this performance flaw. Pushing a list of full paths and base-names maybe a bit more efficient than the current system, but it still leaves a lot on the table.

Reply:
Runtime does not fundamentally require all paths on startup. All that the runtime does with the paths during startup is to store them in a hashtable with simple name as the key. 

Maintaining this hashtable on the host side would open a number of opportunities for memory optimizations. For example, we do not need to materialize all full paths.

Reply:
> this hashtable

This hashtable needs to be case-insensitive (assembly binding is case-insensitive). Case-insensitive hashing and comparisons are tricky. The host does not seem to have the infrastructure for it.

So maybe the best design is to push the simple assembly names from the host eagerly, let the runtime maintain the hash table, but then use a callback to resolve the simple name into a full path. It would avoid potential mismatches in case-insensitive behavior for non-ascii file names, etc.